apps/condo/domains/banking/schema/BankAccount.test.js

```diff
@@ -0,0 +1,305 @@
+
+                expect(createdObj.id).toEqual(updatedObj.id)
+                expect(createdObj.approvedAt).toBeNull()
+                expect(createdObj.approvedBy).toBeNull()
+                expect(updatedObj.approvedBy.id).toEqual(admin.user.id)
+            })
+
+            test('when admin updates, approved fields drop', async () => {
+                const admin = await makeLoggedInAdminClient()
+                const [organization] = await createTestOrganization(admin)
+
+                const [ createdObj ] = await createTestBankAccount(admin, organization, { approvedBy: { connect: { id: admin.user.id } } })
+                const [ updatedObj ] = await updateTestBankAccount(admin, createdObj.id, { bankName: 'NewBankName' })
+
+                expect(createdObj.id).toEqual(updatedObj.id)
+                expect(createdObj.approvedBy.id).toEqual(admin.user.id)
+                expect(updatedObj.approvedBy).toBeNull()
+                expect(updatedObj.bankName).toEqual('NewBankName')
+            })
+
+            test('support can', async () => {
+                const support = await makeClientWithSupportUser()
+                const [organization] = await createTestOrganization(support)
+
+                const [ createdObj ] = await createTestBankAccount(support, organization)
+                const [ updatedObj ] = await updateTestBankAccount(support, createdObj.id, { bankName: 'NewBankName' })
+
+                expect(createdObj.id).toEqual(updatedObj.id)
+                expect(updatedObj.bankName).toEqual('NewBankName')
+            })
+
+            test('support can update approved fields', async () => {
+                const support = await makeClientWithSupportUser()
+                const [organization] = await createTestOrganization(support)
+
+                const [ createdObj ] = await createTestBankAccount(support, organization)
+                const [ updatedObj ] = await updateTestBankAccount(support, createdObj.id, { approvedBy: { connect: { id: support.user.id } } })
+
+                expect(createdObj.id).toEqual(updatedObj.id)
+                expect(createdObj.approvedAt).toBeNull()
+                expect(createdObj.approvedBy).toBeNull()
+                expect(updatedObj.approvedBy.id).toEqual(support.user.id)
+            })
+
+            test('user can\'t', async () => {
+                const admin = await makeLoggedInAdminClient()
+                const user = await makeClientWithNewRegisteredAndLoggedInUser()
+                const [organization] = await createTestOrganization(admin)
+
+                const [ createdObj ] = await createTestBankAccount(admin, organization)
+
+                await expectToThrowAccessDeniedErrorToObj(async () => {await updateTestBankAccount(user,
+                    createdObj.id, { bankName: 'NewBankName' })
+                })
+            })
+
+            test('user can\'t update isApproved field', async () => {
```

Comment:
Please, write also a test for case when user is trying to update another fields, not only related to approval